Sources/TuistKit/Services/GenerateService.swift

```diff
@@ -54,7 +55,41 @@ final class GenerateService {
             ignoreBinaryCache: ignoreBinaryCache,
             cacheStorage: cacheStorage
         )
-        let (workspacePath, _, environment) = try await generator.generateWithGraph(path: path)
+
+        // Load the graph first to get the total number of projects
+        let graph = try await generator.load(path: path)
```

Comment:
I wonder if this sort of defeats the purpose – loading the graph is usually the time-intensive part and generating the project usually happens really fast, so I'd aim to combine this into a single progress step.

I would consider doing the following:
- Combine loading and generation into a single [progress step](https://noora.tuist.dev/components/step/progress)
- The loading steps should be:
  - "Loading the graph"
  - "Generating the graph"
  - "Generated graph with x projects" or "Generated x projects"

What do you think?

Reply:
I very much agree. Also note that the steps include the time on completion, so we don't need an extra step that says how much it took to run something.